google-cloud-error_reporting/lib/google/cloud/error_reporting.rb

```diff
@@ -0,0 +1,259 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google-cloud-error_reporting"
+require "google/cloud/error_reporting/project"
+require "google/cloud/error_reporting/middleware"
+require "stackdriver/core"
+
+module Google
+  module Cloud
+    ##
+    # # ErrorReporting
+    #
+    # Stackdriver Error Reporting counts, analyzes and aggregates the crashes in
+    # your running cloud services. The Stackdriver Error Reporting
+    # Instrumentation client provides
+    # [a simple way to report errors](#report-error) from your application.
+    #
+    # For general information about Stackdriver Error Reporting, read
+    # [Stackdriver Error Reporting Documentation]
+    # (https://cloud.google.com/error-reporting/docs/).
+    #
+    # The goal of google-cloud-ruby is to provide an API that is comfortable to
+    # Rubyists. Authentication is handled by Google::Cloud#error_reporting.
+    # You can provide the project and credential information to connect to the
+    # Stackdriver Error Reporting service, or if you are running on Google
+    # Cloud Platform this configuration is taken care of for you. You can read
+    # more about the options for connecting in the [Authentication
+    # Guide](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/guides/authentication).
+    #
+    # ## Report Error
+    #
+    # You can easily report exceptions from your applications to Stackdriver
+    # Error Reporting service:
+    #
+    # ```ruby
+    # require "google/cloud/error_reporting"
+    #
+    # # Configure Stackdriver ErrorReporting instrumentation
+    # Google::Cloud::ErrorReporting.configure do |config|
+    #   config.project_id = "my-project"
+    #   config.keyfile = "/path/to/keyfile.json"
+    # end
+    #
+    # # Insert a Rack Middleware to report unhanded exceptions
+    # use Google::Cloud::ErrorReporting::Middleware
+    #
+    # # Or explicitly submit exceptions
+    # begin
+    #   fail "Boom!"
+    # rescue => exception
+    #   Google::Cloud::ErrorReporting.report exception
+    # end
+    # ```
+    #
+    # See the [Instrumentation
+    # Guide](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-error_reporting/guides/instrumentation)
+    # for more examples.
+    #
+    module ErrorReporting
+      ##
+      # @private Instrumentation library configuration options
+      CONFIG_OPTIONS = %I{
+        project_id
+        keyfile
+        service_name
+        service_version
+      }
+
+      # Initialize :error_reporting as a nested Configuration under
+      # Google::Cloud if haven't already
+      unless Google::Cloud.configure.option? :error_reporting
+        Google::Cloud.configure.add_options [{
```

Comment:
Consider allowing add_options to take a simple hash as well, so we don't have this awkwardness here.